tests/test_node.py

```diff
@@ -71,3 +72,46 @@ def test_run_as_root(
                     if record.levelno == logging.DEBUG:
                         found_expected_error = True
             assert found_expected_error
+
+    # An arbitrary time slice that is longer than it takes to spin up a cluster
+    # but avoids bash infinite loops.
+    @pytest.mark.timeout(60 * 15)
+    def test_popen_as_root(
+        self,
+        cluster_backend: ClusterBackend,
+        oss_artifact: Path,
+    ) -> None:
+        """
+        It is possible to run commands as root asynchronously
+        """
+        with Cluster(
+            agents=0,
+            public_agents=0,
+            cluster_backend=cluster_backend,
+            generate_config_path=oss_artifact,
+        ) as cluster:
+            (master, ) = cluster.masters
+            popen_1 = master.popen_as_root(
+                args=[
+                    '(mkfifo', '/tmp/pipe', '|', 'true)'
+                    '&&', '(cat', '/tmp/pipe)'
+                ]
+            )
+            sleep(1)
```

Comment:
I don't understand why there needs to be a sleep - can this be removed? Or a comment added.